src/pointer-analysis/value_set_dereference.cpp

```diff
@@ -61,6 +61,43 @@ static bool should_use_local_definition_for(const exprt &expr)
   return false;
 }
 
+static void do_display_points_to_sets(
+  const messaget &log,
```

Comment:
And another small quibble around the architecture here... Should this function take a `messaget`, or should it return a `json_result` and leave it to the caller to do what it will with that? Obviously this choice has an impact on the naming of the function. In the implementation here, this function has two purposes - formatting json, then writing it to a (provided) output destination. Alternatively, it could simply focus on the formatting aspect, and then make the caller just do `log.status() << this_function(blah)` 
Again, I'm honestly not sure which is the best approach, and I'm happy with either - just floating it as an option to consider.